### PR TITLE
Master depth calculation

### PR DESCRIPTION
**Pull request by gfcapalbo**
Before building we run an update_depth script.

This is made to keep depths in waft updated.
If build is on it's first run the check will not happen. 

Deployed and testing on our internal nightly instance.

### CODE REVIEW

File: bin/update_depths.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+            stdout=PIPE,
+            stderr=PIPE,
+        )
+        .stdout.decode("utf-8")
+        .replace("\n", "")
+    )
+    return int(mindepth)
+
+
+def main():
```
Comment (gjotten) [inline bin/update_depths.py:149]: @gfcapalbo somewhere along the way the merge depth script seems to be deleting the empty lines in our repos.yaml file, messing with human readability:

image

**Review by gjotten — CHANGES_REQUESTED** [review]

File: bin/update_depths.py (hunk 1 above)
Comment (gfcapalbo) [reply bin/update_depths.py:174]: @gjotten Done. now all top level lines have a space before, with the happy side effect of removing duplicate lines . looks like this:

`odoo:
  defaults:
    depth: 108
  remotes:
    origin: https://github.com/OCA/OCB.git
    gfcapalbo: https://github.com/gfcapalbo/Odoo
  target: origin ${ODOO_VERSION}
  merges:
  - origin ${ODOO_VERSION}
  - gfcapalbo 14.0-backport-null-first-query-from-master
  - gfcapalbo 14.0-compute_notify_recipients-query-fix

therp/oe_bvtherp:
  defaults:
    depth: ${WAFT_DEPTH_MERGE}
  remotes:
    origin: git@gitlab.therp.nl:therp/oe_bvtherp.git
  merges:
  - origin ${ODOO_VERSION}
  - origin 14.0-#9929-portal_from_draft_to_discuss
  - origin 14.0-9964-use-web_widget_html_markdown

OCA/account-analytic:
  defaults:
    depth: ${WAFT_DEPTH_MERGE}
  remotes:
    origin: https://github.com/OCA/account-analytic
  merges:
  - origin ${ODOO_VERSION}

OCA/account-financial-reporting:
  defaults:
    depth: ${WAFT_DEPTH_MERGE}
  remotes:
    origin: https://github.com/OCA/account-financial-reporting
  merges:
  - origin ${ODOO_VERSION}

OCA/account-financial-tools:
  defaults:
    depth: 971
  remotes:
    origin: https://github.com/OCA/account-financial-tools
    ddejong: https://github.com/ddejong-therp/account-financial-tools
    gfcapalbo: https://github.com/gfcapalbo/account-financial-tools
  merges:
  - origin ${ODOO_VERSION}
  - ddejong ${ODOO_VERSION}-account_move_default_journal
  - gfcapalbo ${ODOO_VERSION}-error_management_chart_update

OCA/account-fiscal-rule:
  defaults:
    depth: ${WAFT_DEPTH_MERGE}
                                       `

File: build [hunk 2]
```diff
@@ -28,7 +28,8 @@ fi
 pip install --quiet "setuptools<58" && \
 pip install --quiet "pip>=20.3" && \
 (pip uninstall --yes -r "${ODOO_WORK_DIR}/requirements-remove-default.txt" 2>/dev/null || /bin/true) && \
-pip install --quiet --exists-action w -r "${ODOO_WORK_DIR}/requirements-default.txt" && \
+pip install  --exists-action w -r "${ODOO_WORK_DIR}/requirements-default.txt" && \
```
Comment (Hussam-Suleiman) [inline build:31]: @gfcapalbo why did you remove '--quiet'?

Comment (gfcapalbo) [reply build:31]: restoring --quiet. i do not need that anymore.

Comment (Hussam-Suleiman) [reply build]: I will merge this pull request for now, and in same time I am trying to upgrade it to the new version https://github.com/sunflowerit/waftlib/blob/v.23.08.03/.python-files/waft-update-depths .

**Comment by gfcapalbo** [comment]
@gjotten  @Hussam-Suleiman requested changes  done

**Comment by Hussam-Suleiman** [comment]
for now, I will merge the changes to master, and in same time I am working to upgrade it to the new version https://github.com/sunflowerit/waftlib/blob/v.23.08.03/.python-files/waft-update-depths .

**Comment by Hussam-Suleiman** [comment]
@gfcapalbo it does not work for odoo 8 with python 2.7, did you test it with all odoo versions?
image

**Comment by Hussam-Suleiman** [comment]
after I merged this changes and then removed it from master again, I can't open this pull request again, @gfcapalbo do you have any idea?

**Comment by Hussam-Suleiman** [comment]
@gfcapalbo @gjotten @NL66278 sorry for destroyed this pull request, we will continue on https://github.com/sunflowerit/waftlib/pull/40

File: bin/update_depths.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+    prog="get_pins",
+    description="Fetch current pins of this instance",
+    epilog="Outputs pinned repo.yaml",
+)
+
+parser.add_argument("--only-base")
+args = parser.parse_args()
+
+
+all_commits = {}
```
Comment (ddejong-therp) [inline bin/update_depths.py:35]: I see that this is never set?

**Comment by gfcapalbo** [comment]
@Hussam-Suleiman replaced with MR:  https://github.com/sunflowerit/waftlib/pull/41

I have added an exit , if python version is not >= 3.5.
Script will just add a line "Depth update not supported for python < 3.5".

it was easier to manage it this way instead of writing a lot of different code to support python 2.7 wich is deprecated.

**Comment by gfcapalbo** [comment]
@Hussam-Suleiman re-tested.  now script will exit when python is < 3.5 and the depth calculation will not be supported for older projects.... the error you had will not appear anymore.

**Pull request merged**